src/foam/nanos/logger/LogMessage.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.logger',
+  name: 'LogMessage',
+
+  properties: [
+    {
+      name: 'date',
+      class: 'String'
```

Comment:
I feel like "date" shouldn't be a formatted string but rather just an epoch timestamp that we can format at display time when needed.

If I recall formatting timestamps can take a surprising amount of processing, also a formatted timestamp in your current format is 23 characters long, whereas an epoch timestamp to milisecond precision is only 13 characters (would be even smaller if we used binary encoding)

Reply:
This is a replacement for our traditional logging, which of course formats the time, and I think from a developer/troubleshooting standpoint, formatted dates are very useful.  Regarding processing time - we'll be replacing like with like.  We could provide flags for toggling formatting  so in the dev/demo environments we get formatted dates in production we get epoch.